### PR TITLE
[iOS][Android]Added PushWoosh tags initialization

### PR DESCRIPTION
**Pull request by rokuz**
https://jira.mail.ru/browse/MAPSME-2747

### CODE REVIEW

File: android/jni/com/mapswithme/platform/MarketingService.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include "platform/marketing_service.hpp"
+#include "Platform.hpp"
```
Comment (ygorshenin) [inline android/jni/com/mapswithme/platform/MarketingService.cpp]: Please, insert a blank line here.

File: platform/marketing_service.hpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+// Events.
+char const * const DownloaderMapActionFinished = "Downloader_Map_action_finished";
+char const * const SearchEmitResultsAndCoords = "searchEmitResultsAndCoords";
+char const * const RoutingCalculatingRoute = "Routing_CalculatingRoute";
+char const * const BookmarksBookmarkAction = "Bookmarks_Bookmark_action";
+char const * const PlacepageHotelBook = "Placepage_Hotel_book";
+char const * const EditorAddStart = "EditorAdd_start";
+char const * const EditorEditStart = "EditorEdit_start";
+
+} // Marketing
```
Comment (ygorshenin) [inline platform/marketing_service.hpp]: 2 spaces before comment

File: platform/marketing_service.hpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class MarketingService
+{
+public:
+
+  using TPushWooshSenderFn = function<void(string const & tag, vector<string> const & values)>;
```
Comment (ygorshenin) [inline platform/marketing_service.hpp]: We decided to get rid of these ugly "T" prefixes in typedefs.

File: platform/marketing_service.hpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  using TMarketingSenderFn = function<void(string const & tag, map<string, string> const & params)>;
+
+  MarketingService() = default;
```
Comment (ygorshenin) [inline platform/marketing_service.hpp]: Do we need to declare ctor as default?

File: platform/marketing_service.hpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "std/function.hpp"
+#include "std/map.hpp"
+#include "std/string.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace Marketing
```
Comment (ygorshenin) [inline platform/marketing_service.hpp]: Why not to lower-case the namespace name?

File: iphone/Maps/Classes/MapViewController.mm [hunk 6]
```diff
@@ -309,6 +309,9 @@ - (void)showWelcomeScreenIfNeeded
   NSUserDefaults * ud = [NSUserDefaults standardUserDefaults];
   if ([ud boolForKey:[welcomeClass udWelcomeWasShownKey]])
     return;
+  
+  if (isFirstSession)
```
Comment (igrechuhin) [inline iphone/Maps/Classes/MapViewController.mm]: В файле MapsAppDelegate.mm есть метод firstLaunchSetup.
Этому коду место там.

File: map/framework.cpp [hunk 7]
```diff
@@ -2279,14 +2281,35 @@ void Framework::BuildRoute(m2::PointD const & finish, uint32_t timeoutSec)
...
+      tag = isP2P ? Marketing::RoutingP2PVehicleDiscovered : Marketing::RoutingVehicleDiscovered;
+      break;
+    case RouterType::Pedestrian:
+      tag = isP2P ? Marketing::RoutingP2PPedestrianDiscovered : Marketing::RoutingPedestrianDiscovered;
+      break;
+    case RouterType::Bicycle:
+      tag = isP2P ? Marketing::RoutingP2PBicycleDiscovered : Marketing::RoutingBicycleDiscovered;
+      break;
+    case RouterType::Taxi:
+      tag = isP2P ? Marketing::RoutingP2PUberDiscovered : Marketing::RoutingUberDiscovered;
```
Comment (igrechuhin) [inline map/framework.cpp]: Taxi?

File: android/jni/com/mapswithme/util/statistics/PushwooshHelper.cpp [hunk 8]
```diff
@@ -0,0 +1,26 @@
+#include "../../core/jni_helper.hpp"
+
+#include "platform/platform.hpp"
+
+extern "C"
+{
+
```
Comment (ygorshenin) [inline android/jni/com/mapswithme/util/statistics/PushwooshHelper.cpp]: Please, remove this blank line.

File: android/jni/com/mapswithme/platform/MarketingService.cpp [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#include "platform/marketing_service.hpp"
+
+#include "Platform.hpp"
+
+void MarketingService::SendPushWooshTag(string const & tag)
+{
+  SendPushWooshTag(tag, vector<string>{ "1" });
```
Comment (ygorshenin) [inline android/jni/com/mapswithme/platform/MarketingService.cpp]: No need in spaces in initializer lists, i.e.:
```c++
{"1"}
```
Same below, for ```{value}```.

File: platform/marketing_service.hpp [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+#include "std/function.hpp"
+#include "std/map.hpp"
+#include "std/string.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace marketing
+{
+
```
Comment (ygorshenin) [inline platform/marketing_service.hpp]: No need in this blank line.

File: platform/marketing_service.hpp [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+// Events.
+extern char const * const kDownloaderMapActionFinished;
+extern char const * const kSearchEmitResultsAndCoords;
+extern char const * const kRoutingCalculatingRoute;
+extern char const * const kBookmarksBookmarkAction;
+extern char const * const kPlacepageHotelBook;
+extern char const * const kEditorAddStart;
+extern char const * const kEditorEditStart;
+
```
Comment (ygorshenin) [inline platform/marketing_service.hpp]: No need in this blank line.

**Comment by ygorshenin** [comment]
Generally LGTM, but please, fix minor style issues.

**Comment by alexzatsepin** [comment]
LGTM

**Pull request merged**